### PR TITLE
Port TableViews to U2

### PR DESCRIPTION
**Pull request by shepheb**
Implements #512 .

This scraps the classic, row-scrolling `TableView`. These new models are based on the old `foam.ui.(md.)FlexTableView`, which use the `ScrollView` for native smooth scrolling, and allow adjusting columns.

Both the inheritance/composition graph and the code is simpler than the previous versions.

### CODE REVIEW

File: js/foam/u2/TableRowView.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+CLASS({
+  package: 'foam.u2',
+  name: 'TableRowView',
+  extends: 'foam.u2.View',
+  imports: [
+    'dynamic',
```
Comment (kgrgreer) [inline js/foam/u2/TableRowView.js]: Is imported by Element, so not needed.

File: js/foam/u2/TableView.js [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+  extends: 'foam.u2.View',
+  requires: [
+    'SimpleValue',
+    'foam.u2.ScrollView',
+    'foam.u2.TableRowView',
+  ],
+
+  imports: [
+    'document',
+    'dynamic',
```
Comment (kgrgreer) [inline js/foam/u2/TableView.js]: Not needed.

File: js/foam/u2/TableView.js [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+      }
+    },
+    {
+      type: 'Array',
+      name: 'allProperties_',
+      documentation: 'All the (non-hidden) properties on the model.',
+    },
+    {
+      type: 'Array',
+      name: 'selectedProperties_',
```
Comment (kgrgreer) [inline js/foam/u2/TableView.js]: Maybe a more descriptive name.

File: js/foam/u2/TableView.js [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+          data: this.filteredDAO$Proxy,
+          rowHeight: this.rowHeight,
+          rowView: this.makeRow,
+        });
+      }
+    },
+    {
+      type: 'Array',
+      name: 'colWidths_',
+      lazyFactory: function() {
```
Comment (kgrgreer) [inline js/foam/u2/TableView.js]: Why is this needed?  ArrayProperty already defines factory.  Either remove or comment.

File: js/foam/u2/md/OverlayDropdown.js [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+CLASS({
+  package: 'foam.u2.md',
+  name: 'OverlayDropdown',
```
Comment (kgrgreer) [inline js/foam/u2/md/OverlayDropdown.js]: Add some docs to say what this Model does.

File: js/foam/u2/md/OverlayDropdown.js [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+  extends: 'foam.u2.Element',
+  requires: [
+  ],
+
+  imports: [
+    'document',
+    'dynamic',
```
Comment (kgrgreer) [inline js/foam/u2/md/OverlayDropdown.js]: Not needed.

**Comment by shepheb** [comment]
All addressed. It's now called `columnProperties_`, which is a bit more specific.
